src/Parallel/MemoryMonitor/Tags.hpp

```diff
@@ -0,0 +1,38 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <string>
+#include <unordered_map>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/PrettyType.hpp"
+
+namespace mem_monitor {
+/*!
+ * Gives full subfile path to the dat file for the memory monitor of a
+ * parallel component
+ */
+template <typename ParallelComponent>
+std::string subfile_name() {
+  return "/MemoryMonitors/" + pretty_type::name<ParallelComponent>();
+}
+
+namespace Tags {
+/*!
+ * \brief Tag to hold memory usage of the parallel component it's templated on
+ * before it is written to disk.
+ *
+ * \details The types in the unordered_map are as follows:
+ *
+ * std::unordered_map<Time, std::unordered_map<Node/Proc, Memory in MB>>
+ */
+template <typename ParallelComponent>
```

Comment:
A design decision that's not clear to me is why you opted to have the component be a template parameter instead of using a map with a string as the key. Then you only need one tag in the DataBox instead of many, which simplifies the initialization action quite a bit. You actually wouldn't even need an initialization action anymore, you could trivially use the generic `AddSimpleTags` one. I think you can do that already.

Reply:
When I was writing this, it wasn't clear which components would actually use their tag in the memory monitor and which wouldn't. We always have access to the component we are sizing so it made sense to just template the tag on the component and then grab the tag corresponding to the component when I needed it.

However, with this design, the tags are only really needed for group and nodegroup components so a map of strings probably makes more sense.

Reply:
Okay, is it reasonable to make that change now? Probably before doing so it's worth thinking carefully if there will be any downsides. Nothing jumps out at me since we don't really need the compile time info in the tag, it's basically just used for naming things, and so we can just do the naming dynamically

Reply:
I don't see a downside either. It's basically just making the name a string at runtime vs a template at compile time. Nothing else